base/src/test/java/uk/ac/ebi/atlas/utils/ReactomeClientTest.java

```diff
@@ -1,33 +1,129 @@
-  }
+    @Before
+    public void setUp() throws Exception {
+        subject = new ReactomeClient(restTemplateMock);
+    }
+
+    @Test
+    public void returnEmptyForAnEmptyCollection() {
+        assertThat(subject.getPathwayNames(Collections.emptySet()).entrySet(), hasSize(0));
+    }
+
+    @Test
+    public void skipUnfoundStableIds() {
+        when(restTemplateMock.postForObject(eq(ReactomeClient.URL), anyString(), eq(String.class)))
+                .thenReturn("[" +
+                        "{\"stId\":\"R-FOO-111\", \"displayName\":\"Bar signal\"}," +
+                        "{\"stId\":\"R-FOO-333\", \"displayName\":\"Bar regulator\"}]");
+
+        assertThat(
+                subject.getPathwayNames(ImmutableSet.of("R-FOO-333", "R-FOO-111", "R-FOO-666")).entrySet(),
+                hasSize(2));
+    }
+
+    @Test
+    public void returnEmptyIfSomethingGoesWrong() throws Exception {
+        when(restTemplateMock.postForObject(eq(ReactomeClient.URL), anyString(), eq(String.class)))
+                .thenThrow(new RestClientException("Timeout!"));
+
+        assertThat(subject.getPathwayNames(ImmutableSet.of("R-FOO-123")).entrySet(), hasSize(0));
+    }
+
+    @Test
+    public void returnEmptyIfInvalidResponse() throws Exception {
+        when(restTemplateMock.postForObject(eq(ReactomeClient.URL), anyString(), eq(String.class)))
+                .thenReturn("[{\"stId\":\"R-FOO-111\", \"displayName\":\"Bar signal\"}");
+
+        assertThat(subject.getPathwayNames(ImmutableSet.of("R-FOO-111")).entrySet(), hasSize(0));
+    }
+
+    @Test
+    public void returnEmptyIfEmptyResponse() throws Exception {
+        when(restTemplateMock.postForObject(eq(ReactomeClient.URL), anyString(), eq(String.class))).thenReturn("");
+
+        assertThat(subject.getPathwayNames(ImmutableSet.of("R-FOO-123")).entrySet(), hasSize(0));
+    }
+
+    @Test
+    public void fetchPathways() throws Exception {
+        ImmutableSet<String> pathwayIds = ImmutableSet.of("R-FOO-111", "R-FOO-222", "R-FOO-333");
+
+        when(restTemplateMock.postForObject(eq(ReactomeClient.URL), eq(pathwayIds.stream().collect(Collectors.joining(","))), eq(String.class)))
+                .thenReturn("[" +
+                        "{\"stId\":\"R-FOO-111\", \"displayName\":\"Bar signal\"}," +
+                        "{\"stId\":\"R-FOO-222\", \"displayName\":\"Bar receptor\"}," +
+                        "{\"stId\":\"R-FOO-333\", \"displayName\":\"Bar regulator\"}]"
+                );
+
+        Map<String, String> result = subject.getPathwayNames(pathwayIds);
+
+        assertThat(result.keySet(), hasSize(3));
+        assertThat(result, hasEntry(is("R-FOO-111"), is("Bar signal")));
+        assertThat(result, hasEntry(is("R-FOO-222"), is("Bar receptor")));
+        assertThat(result, hasEntry(is("R-FOO-333"), is("Bar regulator")));
+    }
+
+    @Test
+    public void makesRequestsInBatchesOfQueryMaxSize() throws Exception {
```

Comment:
It's a minor point and this test is not incorrect, but you have done something TDD proponents rail against : you have used a property of an implementation (query 45 terms in batches of 20,20,5) instead of a requirement (query in batches of size no more than 20 because that is the max Reactome will take).